src/atomvm_pico_flash_provider.erl

```diff
@@ -202,17 +202,38 @@ do_flash(ProjectApps, PicoPath, ResetPort) ->
             ok;
         true ->
             Flag = get_stty_file_flag(),
-            Reset = lists:join(" ", [
+            BootselMode = lists:join(" ", [
                 "stty", Flag, ResetPort, "1200"
             ]),
             rebar_api:info("Resetting device at path ~s", [ResetPort]),
-            ResetStatus = os:cmd(Reset),
+            ResetStatus = os:cmd(BootselMode),
             case ResetStatus of
                 "" ->
                     ok;
-                _Any ->
-                    rebar_api:error("Reset ~s failed. Is tty console attached?", [ResetPort]),
-                    rebar_api:abort()
+                Error ->
+                    case os:find_executable(picotool) of
+                        false ->
+                            rebar_api:error("Warning: ~s~nUnable to locate 'picotool', close the serial monitor before flashing, or install picotool for automatic disconnect and BOOTSEL mode.", [Error]),
+                            rebar_api:abort();
+                        _Path ->
+                            rebar_api:warn("Warning: ~s~nFor faster flashing remember to disconnect serial monitor first.", [Error]),
+                            DevReset = lists:join(" ", [
+                                "picotool", "reboot", "-f"
+                            ]),
+                            rebar_api:warn("Disconnecting serial monitor with: `~s' in 5 seconds...", [DevReset]),
+                            timer:sleep(5000),
+                            RebootStatus = os:cmd(DevReset),
+                            case RebootStatus of
+                                "The device was asked to reboot into application mode.\n" ->
+                                    % wait for reboot to settle before attempting to enter BOOTSEL mode
+                                    timer:sleep(3000),
+                                    os:cmd(BootselMode),
+                                    ok;
+                                BootError ->
+                                    rebar_api:error("Failed to prepare pico for flashing: ~s", [BootError]),
+                                    rebar_api:abort()
+                            end
+                    end
             end,
             rebar_api:info("Waiting for the device at path ~s to settle and mount...", [PicoPath]),
             wait_for_mount(PicoPath, 0)
```

Comment:
At the end of this function, while you are at it, can you remove the io:format?  We are already printing the info log the line above, and I think printing to the console (not through the rebar APIs) is maybe not rebar3-endorsed behavior.

Reply:
Yes. That was redundant. I added the `io:format` just in case the `cp` operation failed, I moved that `os:cmd` output into the `rebar_api:info` function, so if something does go wrong at that point, the user will still get the `cp` error message in the log, without the redundant messages.